app/services/session_encryptor.rb

```diff
@@ -8,15 +8,13 @@ def self.build_user_access_key
     build_user_access_key
   end
 
-  # rubocop:disable Security/MarshalLoad
   def self.load(value)
     decrypted = encryptor.decrypt(value, user_access_key)
-    Marshal.load(::Base64.decode64(decrypted))
+    ActiveSupport::JSON.decode(::Base64.decode64(decrypted))
```

Comment:
I thought about that. See my comments in the linked issue. It would be a tradeoff between developer convenience and performance. Not sure it makes sense to call `.with_indifferent_access` on every request.

Also, this is an older version of this file. We don't need to use Base64 encoding and decoding with JSON. This is still a WIP.